scripts/count-assertions.js

```diff
@@ -0,0 +1,133 @@
+const path = require('path');
+const fs = require('fs');
+const csv = require('csv-parser');
+
+const args = require('minimist')(process.argv.slice(2), {
+  alias: {
+    h: 'help',
+  },
+});
+
+if (args.help) {
+  console.log(`
+Default use:
+  node count-assertions.js [directory]
+    Counts test plans, tests, assertions, and command/assertion pairs
+    for all directories. Counts only a single directory if specified as
+    optional first argument.
+  Arguments:
+    -h, --help
+      Show this message.
+`);
+  process.exit();
+}
+
+const testsDirectory = path.resolve(__dirname, '..', 'tests');
+
+let totalTestPlans = 0;
+let totalTests = 0;
+let totalAssertions = 0;
+let totalCommandAssertions = 0;
+
+function parseCSV(filename) {
+  return new Promise((resolve, reject) => {
+    let data = [];
+    fs.createReadStream(filename)
+      .pipe(csv())
+      .on('data', row => {
+        data.push(row);
+      })
+      .on('end', () => {
+        resolve(data);
+      })
+      .on('error', error => {
+        reject(error);
+      });
+  });
+}
+
+// count metrics for tests / commands
+function countTests(tests, commands) {
+  for (let [i, test] of tests.entries()) {
+    const command = commands[i];
+
+    let numCommands = 0;
+    for (let letter of ['A', 'B', 'C', 'D', 'E', 'F']) {
+      if (command[`command${letter}`]) {
+        numCommands++;
+      }
+    }
+
+    let numAssertions = 0;
+    for (let j = 0; j < 7; j++) {
+      if (test[`assertion${j + 1}`]) {
+        numAssertions++;
+      }
+    }
+
+    totalTests++;
+    totalAssertions += numAssertions;
+    totalCommandAssertions += numAssertions * numCommands;
+  }
+}
+
+// count metrics for test plan directory
+function count(directory) {
+  return new Promise((resolve, reject) => {
+    totalTestPlans++;
+
+    parseCSV(path.join(directory, 'data', 'tests.csv'))
+      .then(tests => {
+        parseCSV(path.join(directory, 'data', 'commands.csv'))
+          .then(commands => {
+            countTests(tests, commands);
+            resolve();
+          })
+          .catch(error => {
+            reject(error);
+          });
+      })
+      .catch(error => {
+        reject(error);
+      });
+  });
+}
+
+let promises = [];
```

Comment:
What do you think about wrapping the remaining body of this script in an `async` function?